helper-cli/src/main/kotlin/commands/SetDependencyRepresentationCommand.kt

```diff
@@ -0,0 +1,170 @@
+        .file(mustExist = false, canBeFile = true, canBeDir = false, mustBeWritable = false, mustBeReadable = false)
+        .convert { it.absoluteFile.normalize() }
+
+    private val targetFormat by option(
+        "--format", "-f",
+        help = "The target format for the conversion."
+    ).enum<TargetFormat>().default(TargetFormat.GRAPH)
+
+    override fun run() {
+        val converters = listOf(::convertOrtResult, ::convertAnalyzerResult, ::convertProjectAnalyzerResult)
+        if (converters.none { it() }) {
+            throw UsageError("No supported data type could be found in the input file.")
+        }
+    }
+
+    /**
+     * Convert a full ORT result file.
+     */
+    private fun convertOrtResult(): Boolean =
+        convertResult<OrtResult> { result ->
+            println("Converting a full ORT result file.")
+
+            val analyzerResult = convertToTarget(result.analyzer?.result)
+            val convertedResult = result.copy(
+                analyzer = result.analyzer?.copy(result = analyzerResult)
+            )
+
+            writeResult(convertedResult)
+        }
+
+    /**
+     * Convert an [AnalyzerResult] file.
+     */
+    private fun convertAnalyzerResult(): Boolean =
+        convertResult<AnalyzerResult> {
+            println("Converting an analyzer result file.")
+
+            writeResult(convertToTarget(it))
+        }
+
+    /**
+     * Convert a [ProjectAnalyzerResult]. In this case, the result file contains an [AnalyzerResult], as a
+     * [ProjectAnalyzerResult] does not contain a dependency graph.
+     */
+    private fun convertProjectAnalyzerResult(): Boolean =
+        convertResult<ProjectAnalyzerResult> { result ->
+            println("Converting a project analyzer result file.")
+
+            val analyzerResult = AnalyzerResultBuilder().addResult(result).build()
+            writeResult(convertToTarget(analyzerResult))
+        }
+
+    /**
+     * Convert a result of type [T]. Read this result from the input file and invoke [block] with it. Return *false*
+     * if the result file could not be loaded (which probably means that it contains a result of a different type).
+     */
+    private inline fun <reified T : Any> convertResult(block: (T) -> Unit): Boolean =
+        runCatching {
+            block(ortFile.readValue())
+        }.isSuccess
+
+    /**
+     * Convert the given [analyzerResult] to the selected target format. Fail if no result is available or the
+     * result is already in the target format.
+     */
+    private fun convertToTarget(analyzerResult: AnalyzerResult?): AnalyzerResult {
+        val convertedResult = targetFormat.convert(
+            analyzerResult ?: throw UsageError("No analyzer result available.")
+        )
+
+        if (convertedResult == analyzerResult) {
+            throw UsageError(
+                "The result already uses the dependency graph format. No conversion is done.",
+                statusCode = 2
+            )
+        }
```

Comment:
I'd also prefer to drop this error for same reason as above.

Reply:
e.g. when the command exists successfully, it is ensured that the target file has specified format, no matter what format the input file used. I cannot see any use cases which satisfy the added complexity for these errors.

Reply:
No error is thrown any more.